cmake/OpenCVInstallLayout.cmake

```diff
@@ -0,0 +1,122 @@
+# Following macro updates multiple variables at once:
+#  - OPENCV_BIN_INSTALL_PATH
+#  - OPENCV_LIB_INSTALL_PATH
+#  - OPENCV_CONFIG_INSTALL_PATH
+#  - ...
+macro(ocv_set_install_layout)
+  unset(_var)
+  foreach(arg ${ARGN})
+    if(NOT _var)
+      set(_var ${arg})
+    else()
+      ocv_update(OPENCV_${_var}_INSTALL_PATH ${arg})
+      unset(_var)
+    endif()
+  endforeach()
+endmacro()
+
+if(OPENCV_CUSTOM_INSTALL_LAYOUT)
+
+  include("${OPENCV_CUSTOM_INSTALL_LAYOUT}")
+
+elseif(ANDROID)
+
+  ocv_set_install_layout(
+    BIN            "sdk/native/bin/${ANDROID_NDK_ABI_NAME}"
+    TEST           "sdk/native/bin/${ANDROID_NDK_ABI_NAME}"
+    SAMPLES_BIN    "sdk/native/samples/${ANDROID_NDK_ABI_NAME}"
+    LIB            "sdk/native/libs/${ANDROID_NDK_ABI_NAME}"
+    LIB_ARCHIVE    "sdk/native/staticlibs/${ANDROID_NDK_ABI_NAME}"
+    3P_LIB         "sdk/native/3rdparty/libs/${ANDROID_NDK_ABI_NAME}"
+    CONFIG         "sdk/native/jni"
+    INCLUDE        "sdk/native/jni/include"
+    SAMPLES_SRC    "samples/native"
+    LICENSES       "sdk/etc/licenses"
+    DOC            "doc"
+    TEST_DATA      "sdk/etc/testdata"
+    OTHER          "sdk/etc"
+    JAR            "."
+    JNI            "sdk/native/libs/${ANDROID_NDK_ABI_NAME}"
+    JNI_BIN        "sdk/native/libs/${ANDROID_NDK_ABI_NAME}"
```

Comment:
Do we really need to use short names?

It is really hard to search through file for location of these defines: searching for `OPENCV_JAR_INSTALL_PATH` provides nothing useful.